File: assignment1/pxl3599_lab01.java
/* 
Prabin Lamichhane
1001733599
OS - Windows (tested on omega)
*/
import java.io.File;


public class pxl3599_lab01 {
    // function that gives the size of the directory
    public static long get_dirsize( File dir ) {
        long size = 0;
        File[] all_dir = dir.listFiles(); // list all subdirectories and files inside the main directory
        int dirCount = all_dir.length; // get the count of subdirectories

        for( int i = 0; i < dirCount; i++ ) // loop over all subdirectories
        {
            if( all_dir[i].isFile() ) // check if the subdirectory is a file
            {
                size += all_dir[i].length(); // if yes, get the size of the file
            }
            else // if it is a sub-directory, get the size of the directory by recursively calling this method
            {
                size += get_dirsize( all_dir[i] );
            }
        }
        return size; // return the size of the actual directory
    }
    public static void main(String[] args)
    {
        String dir_name = ".";
        File dir = new File( dir_name );  // access the directory 
        long totalSize = get_dirsize( dir ); // get the size of the directory
        System.out.println(totalSize);
    }
    
}

File: assignment1/pxl3599_lab01.py
""" 
Prabin Lamichhane
1001733599
OS - Windows (tested on omega)
"""
import os

def dir_size(path):
    size = 0
    all_dirs = os.listdir( path ) # access the path, and store the subdirectories in all_dir
    for i_dir in all_dirs: 
        run_dir = os.path.join(path, i_dir) # creating the path to the sub-directory using root directory path
        if os.path.isfile(run_dir): # check if it is a file 
            size += os.path.getsize(run_dir) # get the size of the file and add it to size
        else:
            size += dir_size(run_dir) # else, loop over the files inside subdirectory using recursive function
    return size #return the size at the end

def main():
    print(dir_size("."))

if __name__ == "__main__":
    main()
    